src/pages/Vehicle/Power/models/Tag.ts

```diff
@@ -0,0 +1,56 @@
+import {TagValue} from "./TagValue.ts";
+import {TagType} from "./TagType.ts";
+
+export class Tag {
```

Comment:
What is the purpose of the Tag models? I see you are checking the type of the tag wherever you use it, so the class isn't providing any encapsulation.

Reply:
Precisely Tag is encapsulating all the information related to a tag so that it does not have to be passed separately. It is better to create a Type attribute and use composition than to create several classes, so I check the type when I need to use it.